### PR TITLE
Ui navigation

### PR DESCRIPTION
**Pull request by inyono**

### CODE REVIEW

**Comment by arekkas** [comment]
:+1:

File: src/assets/.jshintrc [hunk 1]
```diff
@@ -22,7 +22,6 @@
     "evil"    : true,
     "forin"   : false,
     "immed"   : true,
-    "laxbreak": false,
```
Comment (arekkas) [inline src/assets/.jshintrc]: Revert these changes?

File: src/assets/source/scripts/modules/serlo_side_navigation.js [hunk 2]
```diff
@@ -287,6 +289,10 @@ define("side_navigation", ["jquery", "underscore", "referrer_history", "events",
                 $ul.append(menuItem.render().$el);
             });
 
+            if (self.levels[0][0].data.parent.data.community) {
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_side_navigation.js]: Looks hacky, because the path is static. Are you sure that this works in every scenario, even when the navigation is deeper than 2 levels?

File: src/assets/source/styles/base/_base.scss [hunk 3]
```diff
@@ -23,18 +23,3 @@ body {
 
   @include theme-font(normal);
 }
-
```
Comment (arekkas) [inline src/assets/source/styles/base/_base.scss]: what did .wrap do?

File: src/assets/source/styles/base/_bootstrap-var-override.scss [hunk 4]
```diff
@@ -32,6 +32,8 @@ $navbarHeight: 25px;
 /* Base Font */
 $sansFontFamily: 'Open Sans', sans-serif;
 
-.btn-block {
```
Comment (arekkas) [inline src/assets/source/styles/base/_bootstrap-var-override.scss]: Reason for removal?

File: src/assets/source/styles/base/_variables.scss [hunk 5]
```diff
@@ -3,21 +3,57 @@
...
-$themeColorLight: lighten($themeColor, 5%);
 
+/**
+ * Logofarbe, wichtigeres Blau, professionell, seriös
+ */
+$serloBlue: #007EC1;
+// $serloBlue: #00A0DF;     //visionsummit-logo blue
+
+/**
+ * "Notfall-Blau" vorerst $serloBlau bevorzugen
```
Comment (arekkas) [inline src/assets/source/styles/base/_variables.scss]: Cool :) Could you maybe translate these into english?

File: src/assets/source/styles/modules/_mixins.scss [hunk 6]
```diff
@@ -14,7 +14,7 @@
   &::-moz-placeholder {
     color: #{$color};
   }
-  &::-ms-placeholder {
```
Comment (arekkas) [inline src/assets/source/styles/modules/_mixins.scss]: Reason for removal?

File: src/assets/source/styles/partials/_serlo_brand.scss [hunk 7]
```diff
@@ -8,31 +8,32 @@
 #serlo-logo {
   font-family: 'serlo';
   font-size: 70px;
+  color: $themeColor;
 }
 
-#subject-nav {
+#header-nav {
```
Comment (arekkas) [inline src/assets/source/styles/partials/_serlo_brand.scss]: Thanks for changing that! :+1:

File: src/assets/source/scripts/modules/serlo_side_navigation.js (hunk 2 above)
Comment (inyono) [inline src/assets/source/scripts/modules/serlo_side_navigation.js]: Yes, it works. `self.levels` contains all levels starting at the dropdown level (i.e. the main-menu does not appear there). So what I basically do, is to grab the first link in the first lvl and check whether its parent (i.e. the main-menu item) is a community link. When you go one level deeper, `self.levels` just adds one entry for the new level, but `self.levels[0]` remains unchanged. So there should be no problem when the dropdown contains at least one item (otherwise it wouldn't be shown anyway)

File: src/module/Ui/templates/layout/navigation/partial/breadcrumb.phtml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+<ol class="breadcrumb">
+<?php
+    foreach ($this->pages as $i => $page) {
+        $html = [];
+        $html[] = '<li>';
+
+        if ($i + 1 < count($this->pages)) {
```
Comment (arekkas) [inline src/module/Ui/templates/layout/navigation/partial/breadcrumb.phtml]: `if ($i + 1 < $pagesCount) {`

and add

`$pagesCount = count($this->pages)`

before foreach

File: src/assets/source/scripts/modules/serlo_side_navigation.js (hunk 2 above)
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_side_navigation.js]: ok :)

File: src/assets/source/styles/modules/_mixins.scss (hunk 6 above)
Comment (inyono) [inline src/assets/source/styles/modules/_mixins.scss]: The right pseudo class is `-ms-input-placeholder` that was added afterwards. `-ms-placeholder` does not exist (Yeah, MS does not use the same name scheme as the others :smiley:)

Comment (arekkas) [inline src/assets/source/styles/modules/_mixins.scss]: :+1:

File: src/assets/source/styles/base/_bootstrap-var-override.scss (hunk 4 above)
Comment (inyono) [inline src/assets/source/styles/base/_bootstrap-var-override.scss]: Already in `src/assets/source/styles/base/_bootstrap-css-override.scss`

Comment (arekkas) [inline src/assets/source/styles/base/_bootstrap-var-override.scss]: :+1: (thanks for cleaning up...)

File: src/assets/source/styles/base/_base.scss (hunk 3 above)
Comment (inyono) [inline src/assets/source/styles/base/_base.scss]: Duplicated code (`src/assets/source/styles/base/_layout.scss`); at least more or less, I have yet to adapt the details

File: src/assets/source/scripts/modules/serlo_breadcrumbs.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    };
+
+    /**
+     * @class Breadcrumbs
+     * @param {Object} options See defaults
+     *
+     * Main constructor
+     **/
+    Breadcrumbs = function (options) {
+        if (!(this instanceof Breadcrumbs)) {
```
Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_breadcrumbs.js]: Shouldn't this be always an instance of Breadcrumbs?

File: src/module/Ui/templates/authentication/activate.twig [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{% do placeholder('hideSubjectNav').set('True') %}
```
Comment (arekkas) [inline src/module/Ui/templates/authentication/activate.twig]: Shouldn't `set(true)` work as well?

**Comment by arekkas** [comment]
Looks good!

File: src/module/Ui/templates/authentication/activate.twig (hunk 10 above)
Comment (inyono) [inline src/module/Ui/templates/authentication/activate.twig]: Yeah, you're right. I feel ashamed that I thought PHP would care what type I pass it ;)

File: src/assets/source/scripts/modules/serlo_breadcrumbs.js (hunk 9 above)
Comment (inyono) [inline src/assets/source/scripts/modules/serlo_breadcrumbs.js]: I was bold and just copied the template of [serlo_side_navigation](https://github.com/serlo-org/athene2/blob/e412973a353fdcd31d02006a35dc5a0c4d0486d9/src/assets/source/scripts/modules/serlo_side_navigation.js). But I think that you don't need the line, even for the singleton pattern (this is already handled by the returned function). Do you remember any reason why it was done in serlo_side_navigation?

Comment (arekkas) [inline src/assets/source/scripts/modules/serlo_breadcrumbs.js]: No idea, but there could be a conflict when using this funciton in a callback without using `.bind(this)`. Probably keep it in there just to be save.

**Comment by inyono** [comment]
It's done! Before deploying: Notice the changes in [serlo.config.local.php.dist](https://github.com/inyono/athene2/blob/ui-navigation/src/config/autoload/serlo.config.local.php.dist)

**Comment by arekkas** [comment]
yeah!

**Comment by arekkas** [comment]
I'll deploy this on friday

**Comment by arekkas** [comment]
Thanks @inyono for the awesome work :)
